### PR TITLE
Unit Details Page

### LINKED ISSUE #28: Simple Process Tracking: Unit Details - Case Manager View
# Overview

## Frontend

Implement the Unit Specific Details - Case Manager view ([Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=286%3A14653&mode=dev)) ignoring the Referrals table at the bottom. You may use dummy data for now until there are dummy test units in the database. Route this page to `/unit/:id` where `:id` is the id of the unit to be currently displayed on the page.

We will implement the Navigation bar and the Referrals section in a later task. 

### Unit API

In  `frontend/src/api/units.ts` create a `getUnit` function that takes in a Unit id as a parameter that calls `GET /api/units/:id` appropriately.

### Functionality

When the page loads, call `getUnit` with the `{id}` parameter from the URL path. Display the returned unit data appropriately on the page.

### Get Unit Route

Implement the `GET /api/units/:id` route

NOTE: You will need to wait for #18 before dummy test units can be created in the database since they are working on the POST /api/units route.

## Acceptance Criteria
- [x] Unit Specific Details - Case Manager view
- [x] `getUnit` API route function
- [x] `GET /api/units/:id` route
- [x] Page displays the correct unit (by `{id}` parameter in URL)

### PR DESCRIPTION
**Pull request by PoliteUnicorn**
## Tracking Info

Resolves #28 

## Changes
We created a unitDetails page and organized it with the styled components. We also created a button so it could be navigated back to the home page. 

## Testing

- I created a dummy page to test that the layout of the page was correct.

## Confirmation of Change
Screenshot 2024-02-11 at 4 15 04 PM

Screenshot 2024-02-11 at 4 15 15 PM

TODO

### CODE REVIEW

File: backend/src/routes/units.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import express from "express";
+const router = express.Router();
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+router.get("/units/:id", (req, res) => {});
```
Comment (petabite) [inline backend/src/routes/units.ts]: Is the Get Unit Route coming in another PR?

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ export default function App() {
           <Routes>
             <Route path="/" element={<Home />} />
             <Route path="/login" element={<Login />} />
+            <Route path="/unit/{id}" element={<UnitDetails />} />
```
Comment (petabite) [inline frontend/src/App.tsx]:
```suggestion
            <Route path="/unit/:id" element={<UnitDetails />} />
```

See [React Router Dynamic Segments docs](https://reactrouter.com/en/main/route/route#dynamic-segments)

File: frontend/src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import styles from "./Button.module.css";
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export interface ButtonProps extends React.ComponentProps<"button"> {
+  label: string;
+  kind?: "primary" | "secondary";
+}
+
+export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(function Button(
```
Comment (petabite) [inline frontend/src/components/Button.tsx]: Andrew created a Button component that's merged to main already. Let's use that instead.

File: frontend/src/pages/Home.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export function Home() {
       <Helmet>
         <title>Home | USHS Housing Portal</title>
       </Helmet>
-      <h1>This is the home page.</h1>
+      <h1>This is the home page. </h1>
```
Comment (petabite) [inline frontend/src/pages/Home.tsx]:
```suggestion
      <h1>This is the home page.</h1>
```

Removing this space should allow this file to merge cleanly with main

File: frontend/src/pages/UnitDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+  `;
+
+  const DoesNotExist = styled.h1`
+    font-size: 48px;
+    font-family: "Neutraface Text", sans-serif;
+    font-weight: 550;
+    line-height: 150%;
+    line-spacing: 0.96px;
+    padding: 20px 10% 20px 10%;
+  `;
```
Comment (petabite) [inline frontend/src/pages/UnitDetails.tsx]: We should move these styled components outside of the component function. Having styled components within the render method actually hurts performance ([explanation from the docs](https://styled-components.com/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method)).

File: frontend/src/pages/UnitDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+          <ListText>{unit.internalNotes}</ListText>
+        </SectionColumn>
+      </Row>
+    </MainColumn>
+  );
+
+  return (
+    <Page>
+      <Helmet>{/* <title>{unit._id} | USHS Housing Portal</title> */}</Helmet>
+      <title>UnitDetails</title>
```
Comment (petabite) [inline frontend/src/pages/UnitDetails.tsx]: ?

File: frontend/src/pages/UnitDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+      <h1>hi</h1>
```
Comment (petabite) [inline frontend/src/pages/UnitDetails.tsx]:
```suggestion
```

**Review by petabite — CHANGES_REQUESTED** [review]
Looking good!! 🎉 Left some comments. Don't forget to fix the merge conflicts.

File: backend/src/routes/units.ts (hunk 1 above)
Comment (JamesZhang878) [reply backend/src/routes/units.ts]: Just to add on to this. Not sure if we want to have the commented section below merged into main

**Review by JamesZhang878 — CHANGES_REQUESTED** [review]
LGTM! Just a couple of things:
1. From my understanding, I should be seeing a page that says that this unit doesn't exist when I try to visit this route? Not sure if this is just an issue on my machine.
Screenshot 2024-02-13 at 7 26 14 PM
2. There were a few places where there is commented out code. Not sure if we want to keep that in when merging.

File: backend/src/routes/units.ts (hunk 1 above)
Comment (PoliteUnicorn) [reply backend/src/routes/units.ts]: Added the get Unit route.

File: frontend/src/components/Button.tsx (hunk 3 above)
Comment (PoliteUnicorn) [reply frontend/src/components/Button.tsx]: Using Andrew's button instead!

**Review by petabite — APPROVED** [review]
Nice work! 🔥

**Comment by petabite** [comment]
@JamesZhang878 Preview URLs for testing:

- https://ushs-housing-portal-staging--pr44-feature-shreya-karen-dsgivjn9.web.app/unit/65cd898003b744c6d21dd423
- https://ushs-housing-portal-staging--pr44-feature-shreya-karen-dsgivjn9.web.app/unit/65d90cb3ff0ee51b335458c1
- https://ushs-housing-portal-staging--pr44-feature-shreya-karen-dsgivjn9.web.app/unit/65d90d2aff0ee51b335458c2

**Review by JamesZhang878 — APPROVED** [review]
LGTM! Very nice work!

**Pull request merged**